selfservice/strategy/oidc/provider_auth0.go

```diff
@@ -0,0 +1,194 @@
+package oidc
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+
+	"github.com/ory/herodot"
+)
+
+type ProviderAuth0 struct {
+	*ProviderGenericOIDC
+}
+
+func NewProviderAuth0(
+	config *Configuration,
+	public *url.URL,
+) *ProviderAuth0 {
+	return &ProviderAuth0{
+		ProviderGenericOIDC: &ProviderGenericOIDC{
+			config: config,
+			public: public,
+		},
+	}
+}
+
+func (g *ProviderAuth0) oauth2() (*oauth2.Config, error) {
+	endpoint, err := url.Parse(g.config.IssuerURL)
+	if err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+
+	authUrl := *endpoint
+	tokenUrl := *endpoint
+
+	authUrl.Path = path.Join(authUrl.Path, "/authorize")
+	tokenUrl.Path = path.Join(tokenUrl.Path, "/oauth/token")
+
+	c := &oauth2.Config{
+		ClientID:     g.config.ClientID,
+		ClientSecret: g.config.ClientSecret,
+		Endpoint: oauth2.Endpoint{
+			AuthURL:  authUrl.String(),
+			TokenURL: tokenUrl.String(),
+		},
+		Scopes:      g.config.Scope,
+		RedirectURL: g.config.Redir(g.public),
+	}
+
+	return c, nil
+}
+
+func (g *ProviderAuth0) OAuth2(ctx context.Context) (*oauth2.Config, error) {
+	return g.oauth2()
+}
+
+func (g *ProviderAuth0) Claims(ctx context.Context, exchange *oauth2.Token) (*Claims, error) {
+	o, err := g.OAuth2(ctx)
+
+	if err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+
+	client := o.Client(ctx, exchange)
+
+	u, err := url.Parse(g.config.IssuerURL)
+	if err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+	u.Path = path.Join(u.Path, "/userinfo")
+	req, err := http.NewRequest("GET", u.String(), nil)
+	if err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+
+	req.Header.Add("Authorization", "Bearer: "+exchange.AccessToken)
```

Comment:
I believe this isn't necessary as o.Client should attach the bearer token automatically :)

Reply:
In my testing, I found that I needed this. I'll take your word for it though.

Reply:
Sorry for the long reply, currently on vacation! If you testing says it does not work without this then you're right, of course :) So, does this only work if it was added?